src/reducers/index.js

```diff
@@ -1004,4 +1008,23 @@ export const onWalletStateChanged = (state, { payload }) => ({
   walletState: payload,
 });
 
+export const onUpdateMiningServer = (state, { payload }) => {
+  const wallet = getGlobalWallet();
+  if (payload.reset) {
+    wallet.storage.config.setTxMiningUrl();
+    LOCAL_STORE.resetMiningServer();
```

Comment:
Why is this on the reducer? I think this is even worse than being on the component... Can you use a saga for side effects?

Reply:
Created the logic on the saga and a separate event on the reducer